src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.cs

```diff
@@ -230,15 +231,23 @@ protected virtual async Task<bool> IsAutomationFunctionAsync()
         private async Task<bool> CheckUserConfirmation(string oldFileName)
         {
             await _projectVsServices.ThreadingService.SwitchToUIThread();
+
+            bool disablePromptMessage = false;
             bool userNeedPrompt = _environmentOptions.GetOption("Environment", "ProjectsAndSolution", "PromptForRenameSymbol", false);
             if (userNeedPrompt)
             {
                 string renamePromptMessage = string.Format(CultureInfo.CurrentCulture, VSResources.RenameSymbolPrompt, oldFileName);
+                string dontShowAgainMessage = string.Format(CultureInfo.CurrentCulture, VSResources.DontShowAgain);
+
+                var result = MessageDialog.Show("Microsoft Visual Studio", renamePromptMessage, MessageDialogCommandSet.YesNo, dontShowAgainMessage,
```

Comment:
I would prefer you add a new method on `IUserNotificationServices` to handle this, and move the call to `MessageDialog.Show` to the implementation of that new method.

Reply:
done